graphql-dgs-codegen-core/src/main/kotlin/com/netflix/graphql/dgs/codegen/generators/kotlin2/Support.kt

```diff
@@ -0,0 +1,91 @@
+/*
+ *
+ *  Copyright 2020 Netflix, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+package com.netflix.graphql.dgs.codegen.generators.kotlin2
+
+import com.squareup.kotlinpoet.TypeName
+import graphql.language.Description
+import graphql.language.Document
+import graphql.language.EnumTypeDefinition
+import graphql.language.ImplementingTypeDefinition
+import graphql.language.InterfaceTypeDefinition
+import graphql.language.NamedNode
+import graphql.language.ObjectTypeDefinition
+import graphql.language.UnionTypeDefinition
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+
+internal val logger: Logger = LoggerFactory.getLogger("com.netflix.graphql.dgs.codegen.generators.kotlin2")
+
+internal data class Field(
+    val name: String,
+    val isScalar: Boolean = false,
+    val type: TypeName,
+    val description: Description?,
+)
+
+/**
+ * Returns a map of interface name to list of field names for all interfaces in the document
+ */
+internal fun Document.interfaceFields(): Map<String, List<String>> {
```

Comment:
Many of these could probably be used as generic GQL support functions, but I didn't see an obvious place with similar things